src/rewrites.jl

```diff
@@ -0,0 +1,139 @@
+# use multiple dispatch to generate code 
+rewrite(command::Command) = rewrite(Val(command.command), command)
+
+function rewrite(::Val{:summarize}, command::Command)
+    gc = generate_command(command; options=[:variables, :ifable, :replace_variables, :single_argument])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    column = extract_variable_references(command.arguments[1])
+    quote
+        $setup
+        Kezdi.summarize($sdf, $column[1]) |> $teardown
+    end |> esc
+end
+
+function rewrite(::Val{:regress}, command::Command)
+    gc = generate_command(command; options=[:variables, :ifable])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    quote
+        $setup
+        reg($sdf, @formula $(arguments[1]) ~ $(sum(arguments[2:end]))) |> $teardown
+    end |> esc
+end
+
+function rewrite(::Val{:generate}, command::Command)
+    gc = generate_command(command; options=[:single_argument, :variables, :ifable, :replace_variables, :vectorize, :assignment])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    target_column = get_LHS(command.arguments[1])
+    LHS, RHS = split_assignment(arguments[1])
+    quote
+        if ($target_column in names($df))
+            ArgumentError("Column \"$($target_column)\" already exists in $(names($df))") |> throw
+        else
+            $setup
+            $local_copy[!, $target_column] .= missing
+            $sdf[!, $target_column] .= $RHS
+            $local_copy |> $teardown
+        end
+    end |> esc
+end
+
+function rewrite(::Val{:replace}, command::Command)
+    gc = generate_command(command; options=[:single_argument, :variables, :ifable, :replace_variables, :vectorize, :assignment])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    target_column = get_LHS(command.arguments[1])
+    LHS, RHS = split_assignment(arguments[1])
+    third_vector = gensym()
+    bitmask = build_bitmask(local_copy, vectorize_function_calls(replace_variable_references(local_copy, command.condition)))
+    quote
+        if !($target_column in names($df))
+            ArgumentError("Column \"$($target_column)\" does not exist in $(names($df))") |> throw
+        else
+            $setup
+            if eltype($RHS) != eltype($sdf[!, $target_column])
+                local $third_vector = Vector{eltype($RHS)}(undef, nrow($local_copy))
+                $third_vector[$bitmask] .= $RHS
+                $third_vector[.!$bitmask] .= $local_copy[!, $target_column][.!$bitmask]
+                $local_copy[!, $target_column] = $third_vector
+            else
+                $sdf[!, $target_column] .= $RHS
+            end
+            $local_copy |> $teardown
+        end
+    end |> esc
+end
```

Comment:
It is not really clear for me why replace has separate `bitmask` definition when `generate_command()` also creates one. It is in setup and should be the same defined here.

Reply:
Would be nice to merge it. Or return the bitmask, too, in the GeneratedCommand. We need to use the negation of the bitmask, as well.